src/AElf.OS.Network.Grpc/GrpcNetworkModule.cs

```diff
@@ -1,22 +1,59 @@
+using System.Collections.Generic;
+using System.Net;
 using AElf.Modularity;
 using AElf.OS.Network.Infrastructure;
+using Grpc.Core;
+using Grpc.Core.Interceptors;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Volo.Abp.Modularity;
 
 namespace AElf.OS.Network.Grpc
 {
     public class GrpcNetworkModule : AElfModule
     {
+        /// <summary>
+        /// Registers the components implemented by the gRPC library.
+        /// </summary>
         public override void ConfigureServices(ServiceConfigurationContext context)
         {
             context.Services.AddSingleton<IAElfNetworkServer, GrpcNetworkServer>();
-            context.Services.AddSingleton<GrpcPeerPool>();
-            context.Services.AddSingleton<IPeerPool, GrpcPeerPool>();
 
-            context.Services.AddSingleton<PeerService.PeerServiceBase, GrpcServerService>();
-            
+            // Internal dependencies
+            context.Services.AddTransient<IPeerDialer, PeerDialer>();
+            context.Services.AddSingleton<GrpcServerService>();
             context.Services.AddSingleton<AuthInterceptor>();
             context.Services.AddSingleton<RetryInterceptor>();
+
+            // setup the server
+            context.Services.AddSingleton<Server>(o =>
```

Comment:
Why not init server in GrpcNetworkServer

Reply:
Initially it was in GrpcNetworkServer, but it created extra dependencies for nothing (the AuthInterceptor) that actually are more related to the set up of the module, based on the options. If an interceptor is optional it will be never be register in the DI and will never be a dependency.